src/app/features/tech-record/components/edit-tech-record-button/edit-tech-record-button.component.ts

```diff
@@ -66,23 +32,31 @@ export class EditTechRecordButtonComponent implements OnInit {
       );
   }
 
+  get isArchived(): boolean {
+    return this.viewableTechRecord?.statusCode === StatusCodes.ARCHIVED;
+  }
+
   getLatestRecordTimestamp(record: VehicleTechRecordModel): number {
     return Math.max(...record.techRecord.map(record => new Date(record.createdAt).getTime()));
   }
 
-  submitTechRecord() {
-    this.submitCheckFormValidity.emit((formValid: boolean) => {
-      if (!formValid) {
-        return;
-      }
-      if (this.hasProvisional && this.isCurrent) {
-        this.store.dispatch(updateTechRecords({ systemNumber: this.systemNumber, oldStatusCode: StatusCodes.PROVISIONAL }));
-      } else if (this.hasProvisional) {
-        this.store.dispatch(updateTechRecords({ systemNumber: this.systemNumber }));
-      } else {
-        this.store.dispatch(createProvisionalTechRecord({ systemNumber: this.systemNumber }));
-      }
+  toggleEditMode() {
+    this.editableState = !this.editableState;
+    this.editableStateChange.emit(this.editableState);
+  }
+
+  cancelAmend() {
+    if (!this.isDirty || confirm('Your changes will not be saved. Are you sure?')) {
       this.toggleEditMode();
-    });
+    }
+
+    this.errorService.clearErrors();
+    this.store.dispatch(updateEditingTechRecordCancel());
+  }
+
+  submitTechRecord() {
+    this.submitCheckFormValidity.emit();
```

Comment:
How come we are removing the logic here? I thought this was needed?

Reply:
Oh this logic has been moved to the parent?

Reply:
Yep